src/test/java/org/prebid/server/bidder/gothamads/GothamAdsBidderTest.java

```diff
@@ -0,0 +1,246 @@
+                .isEqualTo(expectedResults(expectedBidRequest, IMP_ID, secondImpId).getValue());
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingBidsWhenResponseCanNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Bad Server Response"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("responsesWithoutSeatBids")
+    public void shouldReturnErrorWhileMakingBidsWhenResponseDoesNotHaveSeatBid(String response) {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, response);
+
+        // when
+        final Result<List<BidderBid>> actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Empty SeatBid array"));
+    }
+
+    @Test
+    public void shouldReturnBids() {
+        // given
+        final Bid bannerBid = Bid.builder().impid("1").mtype(1).build();
+        final Bid videoBid = Bid.builder().impid("2").mtype(2).build();
+        final Bid nativeBid = Bid.builder().impid("3").mtype(4).build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                givenBidRequest(identity()),
+                givenBidResponse(bannerBid, videoBid, nativeBid)
+        );
+
+        // when
+        final Result<List<BidderBid>> result = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).containsExactly(
+                BidderBid.of(bannerBid, banner, null),
+                BidderBid.of(videoBid, video, null),
+                BidderBid.of(nativeBid, xNative, null)
+        );
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingBidsWhenImpTypeIsNotSupported() {
+        // given
+        final Bid audioBid = Bid.builder().impid("3").mtype(3).build();
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                givenBidRequest(identity()),
+                givenBidResponse(audioBid)
+        );
+
+        // when
+        final Result<List<BidderBid>> result = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors())
+                .containsExactly(badServerResponse("Unable to fetch mediaType 3 in multi-format: 3"));
+    }
+
+    private static Stream<Arguments> responsesWithoutSeatBids() {
+        return Stream.of(
+                Arguments.of(givenEmptyBidResponse()),
+                Arguments.of(givenBidResponse())
+        );
+    }
+
+    private static Result<List<HttpRequest<BidRequest>>> expectedResults(BidRequest bidRequest, String... impIds) {
```

Comment:
Why do you create separate method for single test without generic parameters? Please, inline this, as this approach makes tests harder to read.

Reply:
why do we need generic parameters? I don't understand what you were suggesting